components/bl-autocomplete-component/src/index.js

```diff
@@ -21,10 +21,20 @@ export default function AutocompleteComponent({ component, eventHandlers }) {
     setOptionsList(validate(options));
   }, [options]);
 
+  const hasGroup = optionsList[0]?.hasOwnProperty("groupLabel");
   const autocompleteHeight = autocompleteRef.current?.getBoundingClientRect()?.height;
-  const filteredOptions = optionsList.filter(({ label }) => (
-    label.toLowerCase().includes(inputValue.toLowerCase())
-  ));
+
+  const filteredOptions = optionsList.reduce((acc, item) => {
+    const isMatched = label => label.toLowerCase().includes(inputValue.toLowerCase());
```

Comment:
You create this function on every render `optionList.length` times. Move it out from a component. Also, I could offer another name - compareCaseInsensitive(str1, str2). So you can choose which you prefer more.